### PR TITLE
✨ support singleton in hosted mode

### PR DESCRIPTION
**Pull request by qiujian16**
## Related issue(s)

Fixes #

### CODE REVIEW

File: manifests/klusterlet/management/klusterlet-agent-deployment.yaml [hunk 1]
```diff
@@ -115,3 +124,8 @@ spec:
       - name: hub-kubeconfig
         emptyDir:
           medium: Memory
+      {{if eq .InstallMode "SingletonHosted"}}
+      - name: spoke-kubeconfig-secret
+        secret:
+          secretName: {{ .ExternalManagedKubeConfigWorkSecret }}
```
Comment (zhujian7) [inline manifests/klusterlet/management/klusterlet-agent-deployment.yaml:130]: for singletonHosted mode, will it still generate two(external registration/work) secrets? why use the work secret here?

Comment (qiujian16) [reply manifests/klusterlet/management/klusterlet-agent-deployment.yaml]: yeah, it is a bit confusing. Updated and add some comment.

**Pull request merged**